installer/bundle/bundle_skel_SunOS.sh

```diff
@@ -0,0 +1,432 @@
+
+    if [ $# -ne 2 ]; then
+        echo "INTERNAL ERROR: Incorrect number of parameters passed to getVersionNumber" >&2
+        cleanup_and_exit 1
+    fi
+
+    echo $1 | sed -e "s/$2//" -e 's/\.solaris\..*$//' -e 's/-/./'
+}
+
+verifyNoInstallationOption()
+{
+    if [ -n "${installMode}" ]; then
+        echo "$0: Conflicting qualifiers, exiting" >&2
+        cleanup_and_exit 1
+    fi
+
+    return;
+}
+
+# $1 - The name of the package to check as to whether it's installed
+check_if_pkg_is_installed() {
+    /usr/bin/pkginfo MSFT$1 2> /dev/null 1> /dev/null
+    return $?
+}
+
+# $1 - The filename of the package to be installed
+# $2 - The package name of the package to be installed
+pkg_add() {
+    pkg_filename=$1
+    pkg_name=$2
+
+    echo "----- Installing package: $pkg_name ($pkg_filename) -----"
+    /usr/sbin/pkgadd -a scx-admin -n -d $pkg_filename MSFT$pkg_name
+}
+
+# $1 - The package name of the package to be uninstalled
+# $2 - Optional parameter. Only used when forcibly removing omi on SunOS
+pkg_rm() {
+    echo "----- Removing package: $1 -----"
+    if [ "$2" = "force" ]; then
+        /usr/sbin/pkgrm -a scx-admin-upgrade -n MSFT$1 # 1> /dev/null 2> /dev/null
+    else
+        /usr/sbin/pkgrm -a scx-admin -n MSFT$1 # 1> /dev/null 2> /dev/null
+    fi
+}
+
+
+# $1 - The filename of the package to be installed
+# $2 - The package name of the package to be installed
+# $3 - Okay to upgrade the package? (Optional)
+pkg_upd() {
+    pkg_filename=$1
+    pkg_name=$2
+
+    echo "----- Updating package: $pkg_name ($pkg_filename) -----"
+
+    # No notion of "--force" since Sun package has no notion of update
+    check_if_pkg_is_installed ${pkg_name}
+    if [ $? -eq 0 ]; then
+        # Check version numbers of this package, both installed and the new file
+        INSTALLED_VERSION=`pkginfo -l MSFT${pkg_name} | grep VERSION | awk '{ print $2 }'`
+        FILE_VERSION=`pkginfo -l -d $1 | grep VERSION | awk '{ print $2 }'`
+        IV_1=`echo $INSTALLED_VERSION | awk -F. '{ print $1 }'`
+        IV_2=`echo $INSTALLED_VERSION | awk -F. '{ print $2 }'`
+        IV_3=`echo $INSTALLED_VERSION | awk -F. '{ print $3 }' | awk -F- '{ print $1 }'`
+        IV_4=`echo $INSTALLED_VERSION | awk -F. '{ print $3 }' | awk -F- '{ print $2 }'`
+        FV_1=`echo $FILE_VERSION | awk -F. '{ print $1 }'`
+        FV_2=`echo $FILE_VERSION | awk -F. '{ print $2 }'`
+        FV_3=`echo $FILE_VERSION | awk -F. '{ print $3 }' | awk -F- '{ print $1 }'`
+        FV_4=`echo $FILE_VERSION | awk -F. '{ print $3 }' | awk -F- '{ print $2 }'`
+
+        # If the new version is greater than the previous, upgrade it. We expect at least 3 tokens in the version.
+        UPGRADE_PACKAGE=0
+        if [ $FV_1 -gt $IV_1 -o $FV_2 -gt $IV_2 -o  $FV_3 -gt $IV_3 ]; then
+            UPGRADE_PACKAGE=1
+        elif [ -n "$FV_4" -a -n "$IV_4" ]; then
+            if [ $FV_4 -gt $IV_4 ]; then
+                UPGRADE_PACKAGE=1
+            fi
+        fi
```

Comment:
Holy smokes, I had no idea this was here!

If it works, it works. But I wonder if it's worth while to bring the Linux version comparison code over here as well to deal with Sun's crappy installer?

On the bright side, with all these changes, you'd only focus on testing the Solaris bundle, so you'd get to immediately benefit from the changes! :smile:

It took me a while to get the version comparison code "right" for Linux, and frankly, I'm not 100% sure this is right here. But I'll leave that up to you in the end.